src/Serilog.Extensions.Hosting/SerilogHostBuilderExtensions.cs

```diff
@@ -30,49 +30,102 @@ public static class SerilogHostBuilderExtensions
+                            factory.AddProvider(provider);
+
+                        return factory;
+                    });
+                }
+                else
+                {
+                    collection.AddSingleton<ILoggerFactory>(services => new SerilogLoggerFactory(logger, dispose));
+                }
+            });
+
             return builder;
         }
 
-        /// <summary>
-        /// Sets Serilog as the logging provider.
-        /// </summary>
+        /// <summary>Sets Serilog as the logging provider.</summary>
         /// <remarks>
         /// A <see cref="HostBuilderContext"/> is supplied so that configuration and hosting information can be used.
         /// The logger will be shut down when application services are disposed.
         /// </remarks>
         /// <param name="builder">The host builder to configure.</param>
         /// <param name="configureLogger">The delegate for configuring the <see cref="LoggerConfiguration" /> that will be used to construct a <see cref="Logger" />.</param>
         /// <param name="preserveStaticLogger">Indicates whether to preserve the value of <see cref="Log.Logger"/>.</param>
-        /// <returns>The (generic) host builder.</returns>
-        public static IHostBuilder UseSerilog(this IHostBuilder builder, Action<HostBuilderContext, LoggerConfiguration> configureLogger, bool preserveStaticLogger = false)
+        /// <param name="writeToProviders">By default, Serilog does not write events to <see cref="ILoggerProvider"/>s registered through
+        /// the Microsoft.Extensions.Logging API. Normally, equivalent Serilog sinks are used in place of providers. Specify
+        /// <c>true</c> to write events to all providers.</param>
+        /// <returns>The host builder.</returns>
+        public static IHostBuilder UseSerilog(
+            this IHostBuilder builder,
+            Action<HostBuilderContext, LoggerConfiguration> configureLogger,
+            bool preserveStaticLogger = false,
+            bool writeToProviders = false)
         {
             if (builder == null) throw new ArgumentNullException(nameof(builder));
             if (configureLogger == null) throw new ArgumentNullException(nameof(configureLogger));
             builder.ConfigureServices((context, collection) =>
             {
                 var loggerConfiguration = new LoggerConfiguration();
+
+                LoggerProviderCollection loggerProviders = null;
+                if (writeToProviders)
+                {
+                    loggerProviders = new LoggerProviderCollection();
+                    loggerConfiguration.WriteTo.Providers(loggerProviders);
+                }
+
                 configureLogger(context, loggerConfiguration);
                 var logger = loggerConfiguration.CreateLogger();
+                
+                ILogger registeredLogger = null;
                 if (preserveStaticLogger)
```

Comment:
Hmm something's gone wrong with this upstream, this is the opposite of what `preserveStaticLogger` should do; will fix.. :-)